process_config.sh

```diff
@@ -39,31 +39,8 @@ function print_dev_help
 
 function find_partition_for_dir
 {
-    PART=
-    longest_substr=
-    mount |
-	{ while IFS= read -r var
-	  do
-	      curpart=$(echo "$var" | cut -f 1 -d " ")
-
-	      if [[ "$(echo $curpart | egrep '/')" == "" ]]; then
-		  continue
-	      fi
-
-	      mountpoint=$(echo "$var" | \
-			       sed 's<.* on \(.*\)<\1<' | \
-			       sed 's<\(.*\) type.*<\1<')
-	      substr=$(printf "%s\n%s\n" "$mountpoint" "$1" | \
-			   sed -e 'N;s/^\(.*\).*\n\1.*$/\1/')
-
-	      if [[ "$substr" == $mountpoint && \
-			${#substr} -gt ${#longest_substr} ]] ; then
-		  longest_substr=$substr
-		  PART=$(echo "$var" | cut -f 1 -d " ")
-	      fi
-	  done
-	  echo $PART
```

Comment:
You replace this code with a much more compact line. Yet I had to use this solution because the option --output=fstype is not available in older df versions. And this suite is often used on systems running older software